evalml/tests/automl_tests/test_automl.py

```diff
@@ -2205,6 +2203,87 @@ def test_automl_pipeline_params_multiple(mock_score, mock_fit, X_y_regression):
             assert row['parameters']['Elastic Net Regressor']['l1_ratio'] == Categorical((0.01, 0.02, 0.03)).rvs(random_state=automl.random_seed)
 
 
+@patch('evalml.pipelines.BinaryClassificationPipeline.fit')
+@patch('evalml.pipelines.BinaryClassificationPipeline.score', return_value={"Log Loss Binary": 0.02})
+def test_automl_respects_pipeline_parameters_with_duplicate_components(mock_score, mock_fit, X_y_binary):
+
+    X, y = X_y_binary
+
+    class PipelineDict(BinaryClassificationPipeline):
+        component_graph = {"Imputer": ["Imputer"],
+                           "Imputer_1": ["Imputer", "Imputer"],
+                           "Random Forest Classifier": ["Random Forest Classifier", "Imputer_1"]}
+
+    class PipeLineLinear(BinaryClassificationPipeline):
```

Comment:
I'm glad this PR fixes this case, though I have to say its a bit weird for the user to specify `Imputer_1` in `pipeline_parameters`; I guess this is dependent on them knowing how the naming would work for duplicate cases. Looks fine though!

Reply:
I agree it's weird. One thing I thought about doing was be able to specify custom component names for the list api by using tuples. Something like

```python
class MyPipeline(BinaryClassificationPipeline):
     component_graph = [("My Imputer", "Imputer"), ("Cool Classifier", "Logistic Regression Classifier")]
```

but I thought that too big a change for what the original issue asked for. Plus, not sure how long the list api will be around 😅  I will file an issue so we can decide if this is worth doing and prioritize if so.

Reply:
FYI #2146

Reply:
I put #2146 into the next sprint, per the Elder Council's approval.